Backend/WebAPI/Controllers/ApplicationController.cs

```diff
@@ -0,0 +1,208 @@
+
+namespace WebAPI.Controllers
+{
+    public class ApplicationController : ApiController
+    {
+        /// <summary>
+        /// Application fields received in Register POST request.
+        /// </summary>
+        public class RegisterRequest
+        {
+            public string title;
+            public string url;
+            public string email;
+            public string deleteUrl;
+        }
+
+        /// <summary>
+        /// Application and ApiKey fields received in Publish POST request.
+        /// </summary>
+        public class PublishRequest
+        {
+            public string key;
+            public string title;
+            public string description;
+            public string logo;
+        }
+
+        [HttpPost]
+        [Route("api/applications/register")]
+        public HttpResponseMessage Register([FromBody] RegisterRequest request)
+        {
+            HttpResponseMessage response;
+                                   
+            try
+            {
+                // Check for a valid email.
+                var valid = new System.Net.Mail.MailAddress(request.email);
+            }
+            catch (Exception)
+            {
+                // Error response
+                response = Request.CreateResponse(HttpStatusCode.BadRequest, "Invalid Email");
+                return response;
+            }
+
+            // Check for a valid application url
+            Uri uriResult;
+            bool result = Uri.TryCreate(request.url, UriKind.Absolute, out uriResult)
+                && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps);
+
+            // Check for a valid user deletion url
+            Uri deleteUriResult;
+            bool deleteResult = Uri.TryCreate(request.deleteUrl, UriKind.Absolute, out deleteUriResult)
+                && (deleteUriResult.Scheme == Uri.UriSchemeHttp || deleteUriResult.Scheme == Uri.UriSchemeHttps);
+
+            if (!result) // Invalid application url
+            {
+                // Error response
+                response = Request.CreateResponse(HttpStatusCode.BadRequest, "Invalid Application Url");
+                return response;
+            }
+            else if (!deleteResult) // Invalid user deletion url
+            {
+                // Error response
+                response = Request.CreateResponse(HttpStatusCode.BadRequest, "Invalid User Deletion Url");
+                return response;
+            }
+
+            // Create a new application
+            Application app = new Application
+            {
+                Title = request.title,
+                Url = uriResult.ToString(),
+                Email = request.email,
+                UserDeletionUrl = request.deleteUrl
+            };
+
+            // Create a new ApiKey
+            ApiKey apiKey = new ApiKey
+            {
+                ApplicationUrl = uriResult.ToString()
+            };
+
+            using (var _db = new DatabaseContext())
```

Comment:
A manager class can be used to move all of this logic into its own abstraction layer. This can make the router cleaner and smaller, but will add a bit of redundancy. The response messages can be returned from the manager class

Reply:
Got it.  I'll reorganize.